docs/source/developer-guide/sparse-attention-development.md

```
@@ -0,0 +1,282 @@
+  tokens to attend to during compute).
+- **`AttentionOp`** — consumes those indices via pre/post kernels and
+  drives the core attention kernels. The op already understands
+  page-level sparsity for MQA/GQA in the generation phase, token-level
+  sparsity for MLA in both phases, and token-level KV compression in
+  the context phase for MQA/GQA.
+- **Auxiliary memory subsystem** — manages any extra pools (KT cache,
+  indexer K cache, …) alongside the main KV cache.
+
+<div align="center">
+<figure>
+  <img src="https://github.com/NVIDIA/TensorRT-LLM/raw/main/docs/source/media/sparse_attention_framework.png" width="800">
```

markdownlint-cli2 (0.22.1)

[warning] Images should have alternate text (alt text)

(MD045, no-alt-text)